### PR TITLE
feat: annotation for customizing build timeouts

### PR DESCRIPTION
**Pull request by FenrirUnbound**
## Context

Build timeouts have a default value of 5400 seconds. We want to allow users to override the value. Eventually we will put in efforts to enforce boundaries on these values.

## Objective

Use an annotation for users to configure their own timeouts. Default is 5400 seconds.

## Reference

* Feature issue
   * https://github.com/screwdriver-cd/screwdriver/issues/545
* Hyperctl parameter for build timeout
   * https://github.com/screwdriver-cd/hyperctl-image/pull/19

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,6 +23,7 @@ class K8sExecutor extends Executor {
      * @param  {Object} options.ecosystem.api                         Routable URI to Screwdriver API
      * @param  {Object} options.ecosystem.store                       Routable URI to Screwdriver Store
      * @param  {Object} options.kubernetes                            Kubernetes configuration
+     * @param  {Object} [options.kubernetes.buildTimeout=5400]        Number of seconds to allow a build to run before considering it is timed out
```
Comment (tkyi) [inline index.js]: Object or number?

File: index.js [hunk 2]
```diff
@@ -45,6 +48,7 @@ class K8sExecutor extends Executor {
 
             this.token = fs.existsSync(tokenPath) ? fs.readFileSync(tokenPath).toString() : '';
         }
+        this.buildTimeout = hoek.reach(options, 'kubernetes.buildTimeout') || DEFAULT_BUILD_TIMEOUT;
```
Comment (tkyi) [inline index.js]: can also write this `hoek.reach(options, 'kubernetes.buildTimeout', { default: DEFAULT_BUILD_TIMEOUT });`
not sure what's better

Comment (FenrirUnbound) [reply index.js]: Exceeds the ESLint line width

File: index.js (hunk 1 above)
Comment (FenrirUnbound) [reply index.js]: Should be `Number`. Amended.

File: index.js (hunk 2 above)
Comment (tkyi) [reply index.js]: 🤦‍♀️

Comment (FenrirUnbound) [reply index.js]: IKR 🙄

**Pull request merged**